com.unity.multiplayer.mlapi/Runtime/Core/SnapshotRTT.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using UnityEngine;
+
+namespace MLAPI
+{
+    internal class ClientRtt
+    {
+        /// <summary>
+        /// Round-trip-time data
+        /// </summary>
+        public struct Rtt
+        {
+            public double Best;
+            public double Average;
+            public double Worst;
+            public int SampleCount;
+        }
+        public ClientRtt()
+        {
+            m_RttSendTimes = new double[k_RingSize];
+            m_SendKey = new int[k_RingSize];
+            m_MeasuredLatencies = new double[k_RingSize];
+        }
+
+        /// <summary>
+        /// Returns the Round-trip-time computation for this client
+        /// </summary>
+        public Rtt GetRtt()
```

Comment:
I guess one thing to think about - how many times do you think you might call GetRtt in a given frame?  if you call it a bunch, you might have a dirty flag here and usually return a cached value.  OR, if you need RTT almost every frame and call this multiple times, maybe the computation of the RTT could be in the NotifyAck?  I guess that depends on how often you don't actually need the RTT data.

Reply:
I lean this being "premature optimization" at the moment. I don't want to be cavalier about the feedback, it's good and very valid. But I'd wait until it shows in a profile. With the irons we already have in the fire, I'd wait before adding complexity here. Would that be workable to you?